### PR TITLE
oobmigration: Add additional behavior around versions

### PR DESCRIPTION
**Pull request by efritz**
This PR adds a new functions to the oobmigration package to help with version operations.

## Test plan

New unit tests.

### CODE REVIEW

File: internal/oobmigration/version.go [hunk 1]
```diff
@@ -36,6 +39,42 @@ func (v Version) String() string {
 	return fmt.Sprintf("%d.%d", v.Major, v.Minor)
 }
 
+func (v Version) GitTag() string {
+	return fmt.Sprintf("v%d.%d.0", v.Major, v.Minor)
```
Comment (eseliger) [inline internal/oobmigration/version.go:43]: patch doesn't matter?

File: internal/oobmigration/version.go [hunk 2]
```diff
@@ -36,6 +39,42 @@ func (v Version) String() string {
...
+}
+
+// Next returns the next minor version immediately following the receiver.
+func (v Version) Next() Version {
+	lastMinorVersionInMajorRelease := map[int]int{
+		3: 47, // 3.47.0 -> 4.0.0
```
Comment (eseliger) [inline internal/oobmigration/version.go:49]: todo: adjust

**Review by eseliger — APPROVED** [review]

File: internal/oobmigration/version.go (hunk 1 above)
Comment (efritz) [reply internal/oobmigration/version.go:43]: No, we only track major/minor in oobmigration tables, so you can't actually introduce one in a patch (I think that's explicitly documented but I should double-check).

Also the way we're handing upgrades will be _some version_ -> _minor version, zero patch_, and then you can do the regular update flow to your patch version if you want to (literally just migrate to X.Y.0, then launch X.Y.Z and the migrator will do the small, if any, schema upgrade).

File: internal/oobmigration/version.go (hunk 2 above)
Comment (efritz) [reply internal/oobmigration/version.go]: 2cd875c633fe354157287efba9584e7764c9a680

**Pull request merged**